### PR TITLE
Disallow applicant from accessing tests that are not his

### PR DESCRIPTION
**Pull request by sugrinov**

### CODE REVIEW

File: web/src/main/frontend/src/base/rest/testApi.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+import {get, del, withLoader, withGuestToken} from './requests';
+import {
+  GUEST_TOKEN_KEY,
+  getGuestToken
+} from '../../components/utility/httpUtil';
 
 class testApi {
   getTest(testId) {
-    return withLoader(get(`/tests/${testId}`));
+    return withLoader(withGuestToken(get(`/tests/${testId}`)));
```
[the remark this answers is not in the document]
Comment (sugrinov) [reply web/src/main/frontend/src/base/rest/testApi.js]: Yes, we need to check if test with that id belogs to current applicant

File: web/src/main/frontend/src/base/rest/requests.js [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
-  request.delete(getLink(url)).set('X-Auth-Token', getToken());
+  request.delete(getLink(url)).set(AUTH_TOKEN_KEY, getAuthToken());
 
 export const withLoader = req => {
   LoaderStore.incrementLoader();
   return req
     .then(res => {
       LoaderStore.decrementLoader();
+      if (res.headers['x-guest-token']) {
+        return res;
```
[the remark this answers is not in the document]
Comment (sugrinov) [reply web/src/main/frontend/src/base/rest/requests.js]: This is a workaround for passing headers to `testApi`, since current implementation of `withLoader` only returns response body, and in that case we need headers also.

[the remark this answers is not in the document]
Comment (iscekic) [reply web/src/main/frontend/src/base/rest/requests.js]: Let's make `withLoader` return the entire response and have `get`, `post`, etc. extract the body.

[the remark this answers is not in the document]
Comment (sugrinov) [reply web/src/main/frontend/src/base/rest/requests.js]: Sure, I will make a new PR.

File: web/src/main/frontend/src/components/utility/httpUtil.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const AUTH_TOKEN_KEY = 'X-Auth-Token';
+export const GUEST_TOKEN_KEY = 'X-Guest-Token';
```
Comment (iscekic) [inline web/src/main/frontend/src/components/utility/httpUtil.js]: Why not `const GUEST_TOKEN_KEY = 'x-guest-token'`? HTTP headers are case-insensitive anyway. 😄

Comment (iscekic) [reply web/src/main/frontend/src/components/utility/httpUtil.js]: Reuse token keys for session storage.

File: web/src/main/frontend/src/base/rest/requests.js [hunk 4]
```diff
@@ -1,32 +1,44 @@
...
+} from '../../components/utility/httpUtil';
 
 export const get = url =>
   request
     .get(getLink(url))
-    .set('X-Auth-Token', getToken())
+    .set(AUTH_TOKEN_KEY, getAuthToken())
+    .then(res => res.body);
+
+export const getWithHeaders = (url, headers) =>
```
Comment (iscekic) [inline web/src/main/frontend/src/base/rest/requests.js]: Move `getWithHeaders` and `rawPost` to separate file.

File: web/src/main/frontend/src/base/rest/requests.js [hunk 5]
```diff
@@ -1,32 +1,30 @@
 import request from 'superagent';
 import {getLink} from '../getLink';
 import LoaderStore from '../stores/LoaderStore';
+import {AUTH_TOKEN_KEY, getAuthToken} from '../../components/utility/httpUtil';
 
 export const get = url =>
   request
     .get(getLink(url))
-    .set('X-Auth-Token', getToken())
+    .set(AUTH_TOKEN_KEY, getAuthToken())
```
Comment (iscekic) [inline web/src/main/frontend/src/base/rest/requests.js]: Can we shorten this to `.set(authApi.token)`?

File: web/src/main/frontend/src/base/rest/testApi.js [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
+import {del, withLoader} from './requests';
+import {
+  getGuestToken,
+  GUEST_TOKEN_KEY
+} from '../../components/utility/httpUtil';
 
 class testApi {
   getTest(testId) {
-    return withLoader(get(`/tests/${testId}`));
+    return withLoader(request.get(`/tests/${testId}`).set(testApi.token()));
```
Comment (iscekic) [inline web/src/main/frontend/src/base/rest/testApi.js]: Shouldn't this be `.set(testApi.token)`?

File: web/src/main/frontend/src/base/rest/requests.js (hunk 5 above)
Comment (iscekic) [reply web/src/main/frontend/src/base/rest/requests.js]: Actually, it might create a circular dependency. Just leave it as is if that's the case. 😄

Comment (sugrinov) [reply web/src/main/frontend/src/base/rest/requests.js]: Yup, tried it and it wasn't working :man_juggling:
